src/include/daos/pool_map.h

```diff
@@ -180,6 +178,15 @@ static inline unsigned int pool_buf_nr(size_t size)
 
 struct pool_map;
 
+/**
+ * When domains are passed into gen_pool_buf as an array of integers, the data
+ * should be interpreted as a set of tuples in the form:
+ *   (level number, ID, number of children)
+ * The bottom layer representing ranks is compressed further, in the form:
+ *   (rank)
+ */
+#define POOL_BUF_DOMAIN_TUPLE_LEN	3
```

Comment:
Do you mean domains[0] to domains[3] consist one tuple? Does this rule only apply to "racks", but not to "root", "nodes"? It's a bit subtle. At least we might want to point the reader to gen_pool_buf.

Reply:
The ".proto" file might be a better place to document the format of the array.

Reply:
Domains 0-2 should be one tuple. This applies to root, but not to nodes, which are represented just as a rank number.